src/dBdt.jl

```diff
@@ -32,7 +32,7 @@ This function is the one wrapped by the various integration routines. Based on a
 timepoint `t`, an array of biomasses `biomass`, and a series of simulation
 parameters `p`, it will return `dB/dt` for every species.
 """
-function dBdt(t, biomass, p::Dict{Symbol,Any}, derivative)
+function dBdt(t, biomass, p::Dict{Symbol,Any})
```

Comment:
what was the issue with the pre-allocated version?

Reply:
It prevents the biomasses from varying from one timestep to another, I have removed it for now until I can figure out what is happening. 
I have merged all your other changes but I still can't detect extinction events.